base58/base58.go

```diff
@@ -46,30 +48,43 @@ func Decode(b string) []byte {
 }
 
 // Encode encodes a byte slice to a modified base58 string.
-func Encode(b []byte) string {
-	x := new(big.Int)
-	x.SetBytes(b)
-
-	answer := make([]byte, 0, len(b)*136/100)
-	for x.Cmp(bigZero) > 0 {
-		mod := new(big.Int)
-		x.DivMod(x, bigRadix, mod)
-		answer = append(answer, alphabet[mod.Int64()])
+func Encode(bin []byte) string {
+	binsz := len(bin)
+	var i, j, high, zcount, carry int
+
+	for zcount < binsz && bin[zcount] == 0 {
+		zcount++
 	}
 
-	// leading zero bytes
-	for _, i := range b {
-		if i != 0 {
-			break
+	size := (binsz-zcount)*138/100 + 1
```

Comment:
I wish I knew, but I don't .... 

the implementation is based off this C implementation: https://github.com/trezor/trezor-crypto/blob/master/base58.c
but that has no explanation for it either.